### DISCUSSION TITLE
lab6 sim problem

### DISCUSSION THREAD

**Discussion by sungsungwu** [bol-edu/HLS-SOC-Discussions#153] [Q&A]
不好意思 請問一下此次lab的uart與counter_la_mm在執行source run_sim都會timed out，但是其他兩個在執行時都可以pass，請問這是正常的嗎?這對後續結合起來時會有影響嗎? 感謝!
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/3fdcc0a1-dfd9-4c7b-a44f-7cc05f12b435)
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/b21d59de-142c-4d23-bf91-e1048d7845af)

**Comment by Lanxas1018** [comment]
同學好，我想我可能沒有說明清楚這部分，我這邊補充。
UART內原本的firmware跟testbench我是做一個loop back的測試，讓同學知道說UART是tx/rx正常的，
loop back測試是指說我先從外部(testbench端)send data過去到caravel cpu，當cpu收到資料後，會產生interrupt訊號並進入ISR，
在ISR的firmware中，我將收到的資料寫回去到reg_tx_data，也就是再把剛剛收到的資料輸出到外部(testbench端)，因為我並沒有在testbench寫一個收到什麼樣的訊號後就finish所以程式會一直運行到所設定的clock cycle count然後finish，跟time out沒有關係，同學也可以自己將time out關掉，可以自行去testbench確認我們的行為。

**Reply by sungsungwu** [reply]
好的，感謝詳細的說明，那我再去testbench內查看，謝謝回答!

**Comment by ghost** [comment]
counter_la_mm.c中，最後一行加入reg_mprj_datal = 0xAB510000;就不會timeout了

**Reply by sungsungwu** [reply]
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/113113868/69c639c3-9932-4f15-93c8-548d049f6782)
pass了，感謝!
